invenio_circulation/transitions/base.py

```diff
@@ -142,11 +143,34 @@ def ensure_item_is_available_for_checkout(self, loan):
             raise ItemNotAvailableError(item_pid=loan["item_pid"],
                                         transition=self.dest)
 
+    def ensure_item_is_available_for_checkout(self, loan):
+        """Validate that an item is available."""
+        self._check_item_before_availability(loan)
+
         if not is_item_available_for_checkout(loan["item_pid"]):
             raise ItemNotAvailableError(
                 item_pid=loan["item_pid"], transition=self.dest
             )
 
+    def ensure_at_desk_item_is_available_for_checkout(self, loan):
```

Comment:
since this is used only in `ItemAtDeskToItemOnLoan`, shall we move it inside there and not keep it in the base class?